core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,696 @@ def update_question_suggestion(
+            stat.accepted_questions_without_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionContributionStatsModel.update_timestamps_multi(  # type: ignore[type-var]
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionContributionStatsModel.put_multi(  # type: ignore[type-var]
+        stats_models_to_update)
+
+
+def _update_question_review_stats_models(
+    question_review_stats: List[
+        suggestion_registry.QuestionReviewStats]) -> None:
+    """Updates QuestionReviewStatsModel models for given question
+    review stats.
+
+    Args:
+        question_review_stats: list(QuestionReviewStats). A list of
+            QuestionReviewStats domain objects.
+    """
+    stats_ids = []
+
+    for stat in question_review_stats:
+        stats_ids.append(
+            suggestion_models.QuestionReviewStatsModel.generate_id(
+                stat.contributor_user_id,
+                stat.topic_id
+            ))
+
+    stats_models_to_update = (
+        suggestion_models.QuestionReviewStatsModel.get_multi(stats_ids))
+
+    for index, stats_model in enumerate(stats_models_to_update):
+
+         # Ruling out the possibility of None for mypy type checking.
+        assert stats_model is not None
+        stat = question_review_stats[index]
+        stats_model.reviewed_questions_count = (
+            stat.reviewed_questions_count)
+        stats_model.accepted_questions_count = (
+            stat.accepted_questions_count)
+        stats_model.accepted_questions_with_reviewer_edits_count = (
+            stat.accepted_questions_with_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionReviewStatsModel.update_timestamps_multi(  # type: ignore[type-var]
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionReviewStatsModel.put_multi(  # type: ignore[type-var]
+        stats_models_to_update)
+
+
+def update_translation_contribution_stats(
+    suggestion: suggestion_registry.BaseSuggestion,
+    user_id: str, target_id: str, is_submission: bool) -> None:
+    """Creates/updates TranslationContributionStatsModel model for
+    given translation submitter when a translation is submitted or reviewed.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            submitted or reviewed.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+        is_submission: bool. A boolean value which indicates whether this
+            function is called when submitting a translation or not.
```

Comment:
```suggestion
        is_submission: bool. Indicates whether this function is called when
          submitting a translation.
```

Nit: The arg type is already stated so no need to include in the description as well. Here and below.

Reply:
Done